lms/djangoapps/instructor/views/api.py

```diff
@@ -1151,15 +1190,32 @@ def generate_registration_codes(request, course_id):
         internal_reference=internal_reference,
         customer_reference_number=customer_reference_number
     )
+
+    course = get_course_by_id(course_id, depth=None)
```

Comment:
Ooh. I don't think we want depth=None which will cause the whole course to be fetch/loaded from Mongo. I believe we want depth=0